docs/llms.py

```diff
@@ -0,0 +1,162 @@
+import json
+import os
+import re
+
+EXCLUDE_DIRS = {"_build", ".git", "venv"}
+BASE_DOCS_URL = "https://docs.mlrun.org/en/stable/"
+
+
+def generate_llm_txt(root_dir, prefix="", output_path=None, exclude_dirs=None):
+    """
+    Generates llms.txt by categorizing .md and .ipynb files with extracted titles and descriptions.
+    """
+    exclude_dirs = exclude_dirs or set()
+    markdown_files = find_files_by_extension(
+        root_dir, [".md"], exclude_dirs=exclude_dirs
+    )
+    notebook_files = find_files_by_extension(
+        root_dir, [".ipynb"], exclude_dirs=exclude_dirs
+    )
+
+    parsed_output = []
+
+    # Add Markdown files
+    add_files_to_output(
+        markdown_files,
+        root_dir,
+        parsed_output,
+        "# Documentation",
+        extract_md_first_title,
+        extract_md_first_sentence,
+    )
+
+    # Add Notebook files
+    add_files_to_output(
+        notebook_files,
+        root_dir,
+        parsed_output,
+        "\n## Examples",
+        extract_ipynb_first_title,
+        extract_ipynb_first_sentence,
+    )
+
+    # Determine output path
+    output_path = output_path or os.path.join(root_dir, "llms.txt")
+
+    # Write to file
+    with open(output_path, "w", encoding="utf-8") as file:
+        formatted_output = "\n".join(parsed_output)
+        file.write(f"{prefix}{formatted_output}")
+
+    print(f"✅ Generated llms.txt in {output_path}")
+
+
+def find_files_by_extension(root_dir, extensions, exclude_dirs=None):
+    """
+    Recursively find all files with given extensions, excluding specified directories.
+    """
+    exclude_dirs = exclude_dirs or set()
+    found_files = []
+    for dirpath, dirnames, filenames in os.walk(root_dir):
+        if any(excluded in dirpath for excluded in exclude_dirs):
+            continue  # Skip excluded directories
+
+        for filename in filenames:
+            if any(filename.endswith(ext) for ext in extensions):
+                found_files.append(os.path.join(dirpath, filename))
+
+    return found_files
```

Comment:
Can be simplified if extensions is a tuple

```suggestion
def find_files_by_extension(root_dir, extensions: tuple, exclude_dirs=None):
    """
    Recursively find all files with given extensions, excluding specified directories.
    """
    exclude_dirs = exclude_dirs or set()
    found_files = []
    for dirpath, dirnames, filenames in os.walk(root_dir):
        if any(excluded in dirpath for excluded in exclude_dirs):
            continue  # Skip excluded directories

        for filename in filenames:
            if filename.endswith(extensions):
                found_files.append(os.path.join(dirpath, filename))

    return found_files
```

Reply:
cool suggestion, however I'd not go with it for now as everything was already tested and it's not critical at all in this specific case. Of course, if you don't mind.